### ISSUE TITLE
Show videos of subscription with manual download option

### ISSUE THREAD

**Issue by jo-me** [Tzahi12345/YoutubeDL-Material#254]
I played around with the subscription feature and accidentally downloaded 70GB of stuff in the background without realizing it :)

Is it possible to add a channel and just list its videos without automatically downloading them?

If not, it might be something interesting to enhance the subscription feature.
Each video could have "download video" & "download audio" buttons so that you can manually select which ones to download.

My use case is that there are certain channels with kids songs and I want to select some of them for download without downloading them all or manually putting in the individual video links.

**Comment by Tzahi12345** [comment]
I'm surprised youtube didn't block your IP after all that downloading lol

>Is it possible to add a channel and just list its videos without automatically downloading them?

The short answer is no -- the longer answer is that downloading individual videos isn't really the point of subscriptions, rather it's for downloading existing *and* newly uploaded content. But you do have a great point in that it's not really obvious how to filter out or filter in videos you want. This is possible in theory, but apparently the [functionality is broken](https://github.com/Tzahi12345/YoutubeDL-Material/issues/242).

Once I figure out what's wrong and fix it, you'll be able to filter based on the title by adding the `--reject-title` or `--match-title` args. One possible improvement on this is adding dedicated inputs for filtering, since like I said, those args are not well known.

>Each video could have "download video" & "download audio" buttons so that you can manually select which ones to download.

Doing this is kinda difficult with subscriptions because the channel is downloaded/checked in bulk, the videos are not downloaded individually from the app's perspective. You can see this in action where the subscription's videos won't initially appear until the first download is 100% complete. There is one workaround I just thought of, and that's leveraging the youtube-dl archive system to blacklist videos you don't want. Basically, you can input a text file that contains a list of IDs you want to ignore when downloading, the caveat here being that it only works as a blacklist.

Would the filtering approach be applicable in your case, or would you need finer control, like the blacklist approach?

**Comment by ATM-Tech** [comment]
I agree with this. I'd like to switch to using YouTube-DL Material as my primary place to watch my subscriptions and I plan to delete watched videos after a week. However, the tool automatically downloading all past videos doesn't allow me to do this.

**Comment by Tzahi12345** [comment]
@ATM-Tech You may be able to get around this issue by editing your subscription and disabling "download all uploads." You can then set it to only download videos uploaded in the last week, though you'd have to manually delete older videos. When deleting those old videos, it will say "Delete and redownload", but if it's past the selected time, it won't redownload ([see explanation here](https://github.com/Tzahi12345/YoutubeDL-Material/issues/268#issuecomment-734479654))
